app.py

```diff
@@ -103,33 +106,41 @@
 
 def graph_refresh_solver() -> None:
     """Schedule refresh for packages that are not yet analyzed by solver."""
+    if _THOTH_GRAPH_REFRESH_EAGER_STOP:
+        _LOGGER.info(
+            "Eager stop of scheduling new solver runs for unsolved package versions, packages scheduled: %d",
+            _THOTH_GRAPH_REFRESH_EAGER_STOP,
+        )
+
     indexes = list(_GRAPH_DB.get_python_package_index_urls())
 
     packages = []
     # Iterate over all registered solvers and gather packages which were not solved by them.
     for solver_name in _OPENSHIFT.get_solver_names():
         _LOGGER.info("Checking unsolved packages for solver %r", solver_name)
-        for package_name, version in _GRAPH_DB.retrieve_unsolved_python_packages(solver_name):
-            _LOGGER.info(f"Adding new package %r in version %r", package_name, version)
+        solver_info = _GRAPH_DB.parse_python_solver_name(solver_name)
+        for package_name, version, index_url in _GRAPH_DB.get_unsolved_python_package_versions(
+            os_name=solver_info["os_name"],
+            os_version=solver_info["os_version"],
+            python_version=solver_info["python_version"],
+            count=_THOTH_GRAPH_REFRESH_EAGER_STOP
```

Comment:
`_THOTH_GRAPH_REFRESH_EAGER_STOP` can be set to `None` which will cause issues when performing this query.

https://github.com/thoth-station/storages/blob/a86d088243d4f46f63bd88589edd3e47050dae95/thoth/storages/graph/postgres.py#L976-L978

The `get_unsolved_python_package_versions` method will need to be adjusted so that it accepts `count=None`.

Reply:
`_THOTH_GRAPH_REFRESH_EAGER_STOP = int(os.getenv("THOTH_GRAPH_REFRESH_EAGER_STOP") or 0)` isn't this preventing it?

Reply:
I think it inverts the current logic - nothing is returned from the database (count is zero) as oppose to all entries with the old implementation.

But thinking about this again, we should probably always have some limit not to reach possible serialization issues. What about exposing `_DEFAULT_COUNT` in the `GraphDatabase` adapter and use it as a default eager stopping count?

Reply:
something like `_THOTH_GRAPH_REFRESH_EAGER_STOP = int(os.getenv("THOTH_GRAPH_REFRESH_EAGER_STOP") or _GRAPH_DB._DEFAULT_COUNT)`

Reply:
Yes. Just make the default count public (so it does not start with an underscore).

Reply:
(and if statements around logging can be removed)

Reply:
Thanks! :beers:

Reply:
Updated :)

Reply:
Awesome! Respecting changes can go to the `GraphDatabase` adapter and this can go in. Thanks! 👍

Reply:
I changed the variable to be public in https://github.com/thoth-station/storages/pull/1093